src/components/Postiterie.vue

```diff
@@ -306,22 +305,35 @@ style="shape=image;verticalLabelPosition=bottom;labelBackgroundColor=#ffffff;ver
       xhr.onload = () => {
         if (xhr.response.id){
           
+          //the link to the file on google drive
           this.link = 'https://drive.google.com/file/d/' + xhr.response.id + '/view';
 
-          //if email input, send mail else redirect to the file
-          if (this.email){
-            this.prepareMail();
+          //the link to the file on drawIO (if already allowed on google drive)
+          let profileId = gapi.auth2.getAuthInstance().currentUser.get().getId();
+
+          this.directLink = 'https://www.draw.io/?state={"ids":["' + xhr.response.id + '"],"action":"open","userId":"' + profileId + '"}' + '#G' + xhr.response.id
+
+          //if the user chose to use his gmail email
+          if(this.useGmail){
+              this.email = gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile().U3;
```

Comment:
This kind of chaining is very prone to breaking.
Let's say "currentUser" is undefined, everything will break.

Reply:
I think at this point in the process, the user must be logged into gapi, so this shouldn't be able to break.

Just in case, I've made it so we check if the result is a mail with a simple regex. So that if it isn't, the function can still go on.

Maybe we should add a toast message library to inform the user when such stuff happens ? I would also add something to check if the email field is correct, such things are quick to do with vue.